azure/services/fleetsmember/spec.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleetsmember
+
+import (
+	"context"
+
+	asocontainerservicev1 "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230315preview"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+)
+
+// AzureFleetsMemberSpec defines the specification for azure fleet feature.
+type AzureFleetsMemberSpec struct {
+	Name                 string
+	Namespace            string
+	ResourceGroup        string
+	Group                string
+	SubscriptionID       string
+	ManagerName          string
+	ManagerResourceGroup string
+}
+
+// ResourceRef implements azure.ASOResourceSpecGetter.
+func (s *AzureFleetsMemberSpec) ResourceRef() *asocontainerservicev1.FleetsMember {
+	return &asocontainerservicev1.FleetsMember{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.Name,
+			Namespace: s.Namespace,
+		},
+	}
+}
+
+// Parameters implements azure.ASOResourceSpecGetter.
+func (s *AzureFleetsMemberSpec) Parameters(ctx context.Context, existingFleetsMember *asocontainerservicev1.FleetsMember) (parameters *asocontainerservicev1.FleetsMember, err error) {
+	if existingFleetsMember != nil {
+		return existingFleetsMember, nil
+	}
+
+	fleetsMember := &asocontainerservicev1.FleetsMember{}
+	fleetsMember.Spec = asocontainerservicev1.Fleets_Member_Spec{}
+	fleetsMember.Spec.AzureName = s.ManagerName
+	fleetsMember.Spec.Owner = &genruntime.KnownResourceReference{
+		ARMID: azure.FleetID(s.SubscriptionID, s.ManagerResourceGroup, s.ManagerName),
+	}
+	fleetsMember.Spec.Group = ptr.To(s.Group)
+	fleetsMember.Spec.ClusterResourceReference = &genruntime.ResourceReference{
+		ARMID: azure.ManagedClusterID(s.SubscriptionID, s.ResourceGroup, s.Name),
```

Comment:
is s.ResourceGroup the resourceGroup of the target AKS cluster? or the resource group of the member resource?
I assume that memberSpec.ResourceGroup is the RG of the member?

Reply:
It is the resource group of both the target AKS cluster and the member resource. The member resource is then attached the owning fleet manager. I may be misunderstanding how member resources work though. Are they supposed to be in a different resource group?

Reply:
the member resource is a child resource of the fleet manager, just like an agentpool is a child resource of an aks cluster.
the member cannot be in a different RG than the fleet manager.

the aks cluster that the member resource targets, can be anywhere.

Reply:
>the member cannot be in a different RG than the fleet manager.
Is the `spec.Group` field the resource group of the fleet member then? I don't see any other place where I specified the resource group for the member. If so, I can set Group to `s.ManagerResourceGroup`. For the `ClusterResourceReference`, I think the current logic makes sense as the cluster resource group is separate from the manager/member resource group.

Reply:
`member.Group` is an "UpdateGroup" that the user can provide to group members within a fleet.
"test", "staging", "prod-early", "prod" for example. 
like a static annotation on the member if you will. not a resourcegroup.

Reply:
Gotcha. I just re-tested and it seems like the member cluster I made is in a different resource group than the manager and it seems to be working fine? 
Member cluster:
![image](https://github.com/kubernetes-sigs/cluster-api-provider-azure/assets/47200969/abf2aebd-eb3d-4ecc-82dc-3a2d7f722fc7)
And my fleet manager is in a separate resource group `fleet-test`